packages/client3/components/design-system/wallet-button/wallet-button.tsx

```diff
@@ -1,151 +1,61 @@
-    variables: {
-      userAccount: account?.toLowerCase(),
-      fetchPolicy: "network-only",
-    },
-  });
-  const applicationState = useApplicationState();
-
-  useEffect(() => {
-    const checkWallet = async () => {
-      setIsWrongUser(false);
-      const accountInformation = await checkWalletAddress(
-        account,
-        applicationState
-      );
-      if (accountInformation?.isAccountCorrectState === false) {
-        //set explicitly to false to ignores the `undefined` cases
-        setIsWrongUser(true);
-      }
-    };
-    checkWallet();
-  }, [applicationState, account, error]);
-
-  useEffect(() => {
-    if (user?.data?.user) {
-      setUser && setUser(user.data.user);
-    }
-  }, [user, setUser]);
-
-  const detectBrowser = async () => {
-    const provider = await detectEthereumProvider();
-    if (provider) {
-      return true;
-    }
-    return false;
-  };
-
-  const handleWalletConnect = async () => {
-    setIsLoading(true);
-    if (await detectBrowser()) {
-      if (!isActive) {
-        await metaMask.activate(DESIRED_CHAIN_ID);
-      }
-    }
-    setIsLoading(false);
-  };
-
-  useEffect(() => {
-    if (user?.data?.user) {
-      setUser && setUser(user.data.user);
-    }
-  }, [user, setUser]);
+  const { account, error, connector } = useWallet();
 
-  const getButtonStyles = (): IWalletButtonStyles => {
-    if (isLoading) {
-      return {
-        icon: "",
-        type: ButtonType.PRIMARY,
-        state: ButtonStateText.CONNECTING,
-      };
-    }
-    if (error) {
-      if (error.message.includes("MetaMask not installed")) {
-        return {
-          icon: "Exclamation",
-          type: ButtonType.PRIMARY,
-          state: ButtonStateText.INSTALL_METAMASK,
-          tooltip: "MetaMask not installed",
-        };
-      }
-      return {
-        icon: "Exclamation",
-        type: ButtonType.PRIMARY,
-        state: ButtonStateText.ERROR,
-        tooltip: error.message,
-      };
-    }
-    if (isWrongUser) {
-      return {
-        icon: "Exclamation",
-        type: ButtonType.PRIMARY,
-        state: ButtonStateText.ERROR,
-        tooltip: "Wrong user type",
-      };
-    }
-    if (isActive) {
-      return {
-        icon: "Wallet",
-        type: ButtonType.SECONDARY,
-        state: handleAddressFormat(account as string),
-      };
-    }
-    return {
-      icon: "",
-      type: ButtonType.PRIMARY,
-      state: ButtonStateText.CONNECT,
-    };
-  };
-  const buttonStyles = getButtonStyles();
-
-  return buttonStyles.tooltip ? (
-    <Tooltip content={buttonStyles.tooltip}>
+  return error ? (
+    <Tooltip content={error.message}>
       <Button
-        onClick={handleWalletConnect}
-        isLoading={{ isLoading, position: "left" }}
-        iconLeft={buttonStyles.icon as IconProps["name"]}
-        buttonType={buttonStyles.type as ButtonType}
+        iconLeft="Exclamation"
+        buttonType={ButtonType.PRIMARY as ButtonType}
```

Comment:
Is type casting required? If not can we add the type to the Button component

Reply:
its not, I removed the casting